### ISSUE TITLE
java.io.NotSerializableException: smile.math.kernel.LinearKernel

### ISSUE THREAD

**Issue by crockpotveggies** [haifengl/smile#138]
Went to serialize a trained SVM and got the following error:

```
smile.math.kernel.LinearKernel
java.io.NotSerializableException: smile.math.kernel.LinearKernel
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1184)
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:348)
	at smile.write$.apply(package.scala:33)
	at ai.bernie.SerializationTest$$anonfun$1.apply$mcV$sp(SerializationTest.scala:138)
```

Here's how I'm training/serializing the model:
```
     /*
      Train classifier
     */
    val trainX = trainingImages.map(_._2).toArray
    val trainY = trainingImages.map(_._1).toArray
    val svm = new SVM(new LinearKernel(), 10.0, Math.max(trainY)+1, SVM.Multiclass.ONE_VS_ALL)
    svm.learn(trainX,trainY)
    svm.finish()

    /*
      Serialize model
     */
    smile.write(svm, "/Users/justin/Desktop/serialized_svm.model")

    /*
      Deserialize model
     */
    smile.read("/Users/justin/Desktop/serialized_svm.model").asInstanceOf[SVM[Array[Double]]]
```

Without serialization this model works just fine, looks like a possible bug? It would appear that it's attempting to serialize the `LinearKernel` class instead of the underlying parameters.

**Comment by crockpotveggies** [comment]
Quickly realized that the Scala default `apply()` methods for read/write do not leverage XStream. I've confirmed that by using `smile.write.xstream` I can successfully read/write a model. However, the file size is a bit...big. 11MB is very hefty, is there a way we can just serialize the parameters?

**Comment by haifengl** [comment]
I guess that you are using v1.2.1. This bug is fixed in v1.2.2. Please try the latest version. Thanks!

**Comment by crockpotveggies** [comment]
Confirmed the bug is fixed in v1.2.2. Thanks for your help.
